tests/ha/control_pod/test_control_pod_softfailure.py

```diff
@@ -279,3 +280,166 @@ def test_45498(self):
+        LOGGER.info("Starting background threads for IAM user creation/deletion.")
+        thr_iam.start()
+        for failure in range(HA_CFG["common_params"]["short_loop"]):
+            pod = secrets.SystemRandom().choice(pod_list)
+            LOGGER.info("Creating soft failure of %s for % loop", pod, failure)
+            self.node_master_list[0].kill_process_in_container(pod_name=pod,
+                                                               container_name=const.CORTX_CSM_POD,
+                                                               process_name=const.CSM_AGENT_PRC,
+                                                               safe_kill=True)
+            LOGGER.info("Step 2.1: Check cluster has failures.")
+            resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+            assert_utils.assert_false(resp[0], resp)
+            LOGGER.info("Step 2.1: Checked cluster has failures.")
+        LOGGER.info("Step 2: Started IAM user CRUD operations while performing soft failure on "
+                    "random control pods in loop. Verified existing IAM users persisted.")
+
+        time.sleep(self.csm_soft_delay)
+        LOGGER.info("Step 3: Verify background process for IAM user creation/deletion")
+        event.clear()
+        LOGGER.info("Waiting for background threads to join")
+        thr_iam.join()
+        iam_resp = tuple()
+        while len(iam_resp) != 4:
+            iam_resp = iam_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not iam_resp:
+            assert_utils.assert_true(False, "Background process failed to do IAM user "
+                                            "creation/deletion operations")
+        failed = iam_resp[1]
+        exp_failed = iam_resp[0]
+        user_del_failed = iam_resp[2]
+        created_users = iam_resp[3]
+        assert_utils.assert_false(failed, "No IAM user creation/deletion expected to fail before "
+                                          f"or after soft-failure: {failed}")
+        assert_utils.assert_true(exp_failed, "IAM user creation/deletion expected to fail during "
+                                             f"soft-failure: {exp_failed}")
+        if created_users:
+            for i_i in created_users:
+                self.s3_clean.update(i_i)
+        LOGGER.info("Updating dict for clean up with remaining users")
+        for i_i in user_del_failed:
+            self.s3_clean.update({i_i: del_users[i_i]})
+        LOGGER.info("Step 3: Verified background process for IAM user creation/deletion")
+
+        LOGGER.info("Step 4: Check cluster status is clean.")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        assert_utils.assert_false(resp[0], resp)
+        LOGGER.info("Step 4: Checked cluster has failures.")
+
+        LOGGER.info("Step 5: Get IAM user list and check created IAM users are persist.")
+        resp = self.csm_obj.list_iam_users_rgw()
```

Comment:
Is this API working now? Did you run?
Because earlier it was not working.

Reply:
Yes. Working without any failures